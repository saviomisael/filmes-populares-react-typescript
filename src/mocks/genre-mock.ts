export const genreMock = [{ id: 1, name: 'Action' }];
