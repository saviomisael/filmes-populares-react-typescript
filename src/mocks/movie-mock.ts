export const movieMock = {
  id: 1,
  name: 'Duna',
  rating: 9,
  releaseDate: new Date(),
  poster: '/s9E9W77HS8zEQvsrpz5aEUTKnvD.jpg',
  overview:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
};
