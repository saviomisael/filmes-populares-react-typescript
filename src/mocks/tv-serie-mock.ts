export const tvSerieMock = { id: 1, name: 'Round 6', rating: 7.5, poster: '/6gcHdboppvplmBWxvROc96NJnmm.jpg' };
