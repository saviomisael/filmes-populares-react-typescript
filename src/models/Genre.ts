export class Genre {
  constructor(public readonly name: string) {}
}
