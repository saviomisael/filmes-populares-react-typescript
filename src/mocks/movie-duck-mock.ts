import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { Movie } from 'models/Movie';
import { MovieState } from 'store/ducks/movie-duck';
import { genresMock } from './genre-mock';
import { movieListMock, movieMock } from './movie-mock';

const initialState: MovieState = {
  movies: movieListMock.map((x) => new Movie(x.id, x.name, x.rating, x.releaseDate, x.poster, x.overview)),
  singleMovie: new Movie(
    movieMock.id,
    movieMock.name,
    movieMock.rating,
    movieMock.releaseDate,
    movieMock.poster,
    movieMock.overview,
    [...genresMock],
  ),
};

const movieSlice = createSlice({
  name: 'movie',
  initialState,
  reducers: {
    addMovies: (state, action: PayloadAction<Movie[]>) => {
      state.movies = action.payload;
    },
    updateSingleMovie: (state, action: PayloadAction<Movie>) => {
      state.singleMovie = action.payload;
    },
  },
});

export const { actions: movieMockActions, reducer: movieMockReducer } = movieSlice;
