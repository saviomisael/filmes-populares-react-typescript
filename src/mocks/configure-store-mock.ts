import { configureStore } from '@reduxjs/toolkit';
import { tvSerieMockReducer } from './tv-serie-duck-mock';

export const storeMock = configureStore({
  reducer: {
    tvSeries: tvSerieMockReducer,
  },
});
