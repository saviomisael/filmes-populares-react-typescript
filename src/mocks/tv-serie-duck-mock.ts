import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { TvSerie } from 'models/TvSerie';
import { tvSerieListMock } from 'mocks/tv-serie-mock';

export interface TvSerieState {
  tvSeries: TvSerie[];
}

const initialState: TvSerieState = {
  tvSeries: tvSerieListMock.map((x) => new TvSerie(x.id, x.name, x.rating, x.poster)),
};

const tvSerieSlice = createSlice({
  name: 'tvserie',
  initialState,
  reducers: {
    addSeries: (state, action: PayloadAction<TvSerie[]>) => {
      state.tvSeries = action.payload;
    },
  },
});

export const { actions: tvSerieMockActions, reducer: tvSerieMockReducer } = tvSerieSlice;
