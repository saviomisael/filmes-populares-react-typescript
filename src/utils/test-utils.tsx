import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router-dom';
import { BaseStyles } from 'styles/base';
import { ThemeProvider } from 'styled-components';
import { theme } from 'styles/theme';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { storeMock } from 'mocks/configure-store-mock';

type optionsWithRouter = {
  initialRoutes?: string[];
  pathname: string;
};

export const renderTheme = (children: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <BaseStyles />
      {children}
    </ThemeProvider>,
  );

export const renderWithStoreProvider = (children: ReactNode) =>
  render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <BaseStyles />
        {children}
      </ThemeProvider>
    </Provider>,
  );

export const renderWithRouter = (children: ReactNode, options: optionsWithRouter) => {
  const initialRoutes = options.initialRoutes && options.initialRoutes.length > 0 ? options.initialRoutes : ['/'];

  return render(
    <Provider store={storeMock}>
      <ThemeProvider theme={theme}>
        <MemoryRouter initialEntries={initialRoutes}>
          <Switch>
            <Route path={options.pathname} render={() => children} />
          </Switch>
        </MemoryRouter>
      </ThemeProvider>
    </Provider>,
  );
};
