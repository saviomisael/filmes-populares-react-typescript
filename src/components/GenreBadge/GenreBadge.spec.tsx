import { genreMock } from 'mocks/genre-mock';
import { Genre } from 'models/Genre';
import { renderTheme } from 'utils/test-utils';
import { GenreBadge } from '.';

describe('<GenreBadge />', () => {
  it('should matches with snapshot', () => {
    const genre = new Genre(genreMock[0].id, genreMock[0].name);

    const { container } = renderTheme(<GenreBadge genre={genre} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
