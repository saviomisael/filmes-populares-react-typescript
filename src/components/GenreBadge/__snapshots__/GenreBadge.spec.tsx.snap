// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenreBadge /> should matches with snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  border-radius: 0.6rem;
  padding: 1rem 2rem;
  text-align: center;
  font-weight: 400;
  border: 0.2rem solid #FA9F42;
  font-size: 1.8rem;
  color: #FFFFFF;
}

<div
  class="c0"
>
  <p
    class="c1"
  >
    Action
  </p>
</div>
`;
